crates/fuel-streams-core/src/stream.rs

```diff
@@ -165,28 +172,24 @@ impl<S: Streamable> Stream<S> {
             .is_ok_and(|result| result.is_none())
     }
 
-    /// TODO: investigate why this always returns None even after publishing (putting to the KV Store)
     pub async fn get_last_published(
         &self,
         wildcard: &str,
     ) -> Result<Option<S>, StreamError> {
         let subject_name = &Self::prefix_filter_subject(wildcard);
-
         let message = self
             .store
             .stream
-            .get_last_raw_message_by_subject(subject_name)
+            .direct_get_last_for_subject(subject_name)
```

Comment:
Nice 👍🏾. Did we confirm if `allow_direct` is configured to `true`  by default?

Reply:
It's `true` by default when using kv store
https://github.com/nats-io/nats.rs/blob/b2fb1b406ea9fd2957eadfe2216dea44cd74b56e/async-nats/src/jetstream/context.rs#L731